magpie/owsrequest.py

```diff
@@ -26,31 +26,29 @@ def ows_parser_factory(request):
     Default to JSON if no ``Content-Type`` is specified or if it is 'text/plain' but can be parsed as JSON. Otherwise,
     use the GET/POST WPS parsers.
     """
-    content_type = get_header("Content-Type", request.headers, default=None, split=";,")
-    if content_type is None or content_type == CONTENT_TYPE_PLAIN:
+    content_type = request.headers.get("Content-Type", None)
+
+    def is_json_body(body):
+        if not body:
+            return False
         try:
-            # noinspection PyUnresolvedReferences
-            if request.body:
-                # raises if parsing fails
-                # noinspection PyUnresolvedReferences
-                json.loads(request.body)
-                content_type = CONTENT_TYPE_JSON
+            json.loads(body)
         except ValueError:
-            pass
+            return False
+        return True
+
+    if content_type is None or content_type == CONTENT_TYPE_PLAIN:
+        if is_json_body(request.body):
+            # default to json when can be parsed as json
+            request.content_type = content_type = CONTENT_TYPE_JSON
```

Comment:
not sure if this is requests / pyramid.request compatible...
do both have `.content_type` property?
maybe use `setattr` (to test ?)

Reply:
In `magpie.api.requests.get_multiformat_any`, the `request.content_type` property is accessed. So it's already not `requests` compatible. In which case does this function receive a `requests.Request` object? I would believe it's always a pyramid request.

Reply:
I'm not 100% sure but I remember having multiple places where both where received according to test cases and Twitcher integration. This is why there are multiple util functions to handle both.

Reply:
how about `request.headers["Content-Type"] = request.content_type = content_type = CONTENT_TYPE_JSON`?

Reply:
Probably. 
I have no problem with it if it works as is and all tests pass.